contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,572 @@
+# Raw Block Consumption in Kubernetes
+
+Authors: erinboyd@, screeley44@, mtanino@
+
+This document presents a proposal for managing raw block storage in Kubernetes using the persistent volume source API as a consistent model of consumption.
+
+# Terminology
+* Raw Block Device - a physically attached device devoid of a filesystem
+* Raw Block Volume - a logical abstraction of the raw block device as defined by a path
+* File on Block/Filesystem on Block - a formatted (ie xfs) filesystem on top of a raw block device
+
+# Goals
+* Enable durable access to block storage
+* Support storage requirements for all workloads supported by Kubernetes
+* Provide flexibility for users/vendors to utilize various types of storage devices
+* Agree on API changes for block
+* Provide a consistent security model for block devices 
+* Provide a means for running containerized block storage offerings as non-privileged container
+
+# Non Goals
+* Support all storage devices natively in upstream Kubernetes. Non-standard storage devices are expected to be managed using extension
+  mechanisms.
+* Provide a means for full integration into the scheduler based on non-storage related requests (CPU, etc.)
+* Provide a means of ensuring specific topology to ensure co-location of the data 
+   
+# Value add to Kubernetes
+  
+  By extending the API for volumes to specifically request a raw block device, we provide an explicit method for volume comsumption,
+  whereas previously any request for storage was always fullfilled with a formatted fileystem, even when the underlying storage was 
+  block. In addition, the ability to use a raw block device without a filesystem will allow
+  Kubernetes better support of high performance applications that can utilize raw block devices directly for their storage. 
+  Block volumes are critical to applications like databases (MongoDB, Cassandra) that require consistent I/O performance
+  and low latency. For mission critical applications, like SAP, block storage is a requirement. 
+  
+  For applications that use block storage natively (like MongoDB) no additional configuration is required as the mount path passed
+  to the application provides the device which MongoDB then uses for the storage path in the configuration file (dbpath). Specific
+  tuning for each application to achieve the highest possibly performance is provided as part of its recommended configurations.
+  
+  Specific use cases around improved usage of storage consumption are included in the use cases listed below as follows:
+  * An admin wishes to expose a block volume to be consumed as a block volume for the user  
+  * A user wishes to utilitze block storage to fully realize the performance of an application tuned to using block devices
+  * A user wishes to utilize raw block devices for consumption from a virtual machine
```

Comment:
virtual machines?

Reply:
See http://kubevirt.io or this recording https://asciinema.org/a/134953

We run VMs in pods, and would like to use the volume concept to back virtual disks.